pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -206,25 +211,37 @@ protected void setupInitialSegments(String realtimeTableName, ZNRecord partition
     List<ZNRecord> records = new ArrayList<>(nPartitions);
     final long now = System.currentTimeMillis();
     final int seqNum = 0; // Initial seq number for the segments
+
     for (int i = 0; i < nPartitions; i++) {
-      final List instances = partitionMap.get(Integer.toString(i));
-      LLCRealtimeSegmentZKMetadata metadata = new LLCRealtimeSegmentZKMetadata();
-      final String rawTableName = TableNameBuilder.extractRawTableName(realtimeTableName);
-      LLCSegmentName llcSegmentName = new LLCSegmentName(rawTableName, i, seqNum, now);
-      final String segName = llcSegmentName.getSegmentName();
-
-      metadata.setCreationTime(now);
-      metadata.setStartOffset(startOffset);
-      metadata.setNumReplicas(instances.size());
-      metadata.setTableName(rawTableName);
-      metadata.setSegmentName(segName);
-      metadata.setStatus(CommonConstants.Segment.Realtime.Status.IN_PROGRESS);
-
-      ZNRecord record = metadata.toZNRecord();
-      final String znodePath = PinotRealtimeSegmentManager.getSegmentsPath() + "/" + realtimeTableName + "/" + segName;
-      paths.add(znodePath);
-      records.add(record);
-      idealStateEntries.put(segName, instances);
+      SimpleConsumerWrapper kafkaConsumer = SimpleConsumerWrapper.forPartitionConsumption(new KafkaSimpleConsumerFactoryImpl(),
```

Comment:
maybe simpleconsumerwrapper should take kafkametadata as arg, and not let others worry about extgracting stuff from it. that way, if we have any changes we can modify in metadata and in simpleconsumer without plumbing all code. what do you think?